### PR TITLE
fetch games by userId

### PR DESCRIPTION
**Pull request by mustophy**
I made a very little but necessary tweak to the code we created on video call

I also added a seeder folder containing a mock json for our database in case it gets cleared

### CODE REVIEW

File: server/src/routes/v1/game.route.js [hunk 1]
```diff
@@ -275,7 +275,7 @@ router.patch("/resign", GameCtrl.resign);
 router.get("/:id", GameCtrl.getById);
 
 // Get All Games by User - not implemented -- shauib mustapha
```
Comment (eni4sure) [inline server/src/routes/v1/game.route.js:277]: remove the -- not implemented -- comment

File: server/src/controllers/game.controller.js [hunk 2]
```diff
@@ -358,8 +358,24 @@ class GameController {
...
+          game.owner.user_id == userId ||
+            (game.opponent && game.opponent.user_id == userId) ||
+            (game.spectators?.length > 0 &&  
+              game.spectators.find((spec) => spec.user_id == userId))
+        );
+      });
+
+      return res.status(200).send(response("fetched user games successfully", userGames));
+    } catch(error){
+      next(`Unable to fetch user games ${error}`);
```
Comment (eni4sure) [inline server/src/controllers/game.controller.js:376]: check codes above, we've moved back to the custom error class
kindly fix that 
Thanks

**Review by eni4sure — CHANGES_REQUESTED** [review]

File: server/src/routes/v1/game.route.js [hunk 3]
```diff
@@ -274,8 +274,7 @@ router.patch("/resign", GameCtrl.resign);
  */
 router.get("/:id", GameCtrl.getById);
 
-// Get All Games by User - not implemented -- shauib mustapha
-// router.get("/all/:userId", GameCtrl.getAllByUser);
+router.get("/all/:userId", GameCtrl.getAllByUser);
```
Comment (eni4sure) [inline server/src/routes/v1/game.route.js:277]: still leave the comment
just remove the "not implemented" comment 😅

File: server/src/controllers/game.controller.js [hunk 4]
```diff
@@ -165,7 +165,7 @@ class GameController {
       await centrifugoController.publish(game_id, payload);
       return res.status(200).send(response("pieced moved", updated, true));
     } catch (error) {
-      next(error);
+      throw new CustomError(`Failed to move piece${error}`, 500);
```
Comment (eni4sure) [inline server/src/controllers/game.controller.js:168]: kindly format this error properly - ``piece: ${error}``

**Review by eni4sure — CHANGES_REQUESTED** [review]

File: server/src/routes/v1/game.route.js (hunk 3 above)
Comment (NdubuisiJr) [inline server/src/routes/v1/game.route.js:277]: Write the documentation comment for this route…

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Please add a documentation comment for this route…

**Review by eni4sure — APPROVED** [review]

**Comment by eni4sure** [comment]
CC: @NdubuisiJr kindly approve your change requested is fixed ✔️

**Review by NdubuisiJr — APPROVED** [review]

**Pull request merged**